### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jsturtevant**
Adding a contributing doc which includes guidance for adding and removing new shims.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jprendes) [inline CONTRIBUTING.md:134]:
```suggestion
We welcome new shims, though you can also host them in your own repositories as well and use the `coantainerd-shim-wasm` crate.  We recognize that the project is moving fast and having them in this repository can reduce the need for changes as well for discoverability.  
```

**Review by jprendes — COMMENTED** [review]
Thanks @jsturtevant , this is very needed!

File: CONTRIBUTING.md
Comment (Mossaka) [inline CONTRIBUTING.md:106]: We are actually using `cargo +nightly fmt`

File: CONTRIBUTING.md
Comment (Mossaka) [inline CONTRIBUTING.md:140]: Is ambiguous to me if this is saying the "shim" itself is not maintained, or the "runtime" underneath the shim is not maintainer for 6 months.

File: CONTRIBUTING.md
Comment (jsturtevant) [reply CONTRIBUTING.md:16]: I will make note of this

File: CONTRIBUTING.md
Comment (jsturtevant) [reply CONTRIBUTING.md:136]: cool I will mention we are looking at this and may become a hard requirement in the future.

File: CONTRIBUTING.md
Comment (jsturtevant) [reply CONTRIBUTING.md:140]: I am not sure how to define this.  Open to wording and we can discuss more in the meeting next week.

I don't think it is just the runtime its self (but that is a good thing to call out here).  But also want to avoid a situation if there is a bug in the runtime that surfaces in our test suite or need changes to runtime to support a feature but no one is around to help resolve quickly even if the runtime is actively maintained.

File: CONTRIBUTING.md
Comment (jsturtevant) [reply CONTRIBUTING.md]: updated, please check wording

File: CONTRIBUTING.md
Comment (jsturtevant) [reply CONTRIBUTING.md]: I've taken another shot at this

**Review by jprendes — COMMENTED** [review]
I think we still need to discuss the `Adding new shims` and `Removing shims` sections with stakeholders, but the rest is very valuable technical content.
Can we move those two sections to a follow up PR to unblock the rest of this PR?

**Comment by jsturtevant** [comment]
Move the adding/removing details to #408

**Review by Mossaka — APPROVED** [review]

**Comment by Mossaka** [comment]
Could you please take another look before I merge this in? @jprendes

**Comment by Mossaka** [comment]
Going to merge this in.

**Pull request merged**
